bug: GeometryType fails on all EMPTY geometries

```sql
SELECT GeometryType(st_geomfromtext('GEOMETRYCOLLECTION EMPTY'));


Caused by: java.lang.NullPointerException
        at org.apache.sedona.common.utils.GeomUtils.isMeasuredGeometry(GeomUtils.java:625)
        at org.apache.sedona.common.Functions.geometryTypeWithMeasured(Functions.java:1801)
        at org.apache.spark.sql.sedona_sql.expressions.GeometryType$$anonfun$$lessinit$greater$177.apply(Functions.scala:1736)
        at org.apache.spark.sql.sedona_sql.expressions.GeometryType$$anonfun$$lessinit$greater$177.apply(Functions.scala:1736)
        at org.apache.spark.sql.sedona_sql.expressions.InferrableFunctionConverter$.$anonfun$inferrableFunction1$2(InferrableFunctionConverter.scala:39)
        at org.apache.spark.sql.sedona_sql.expressions.InferredExpression.eval(InferredExpression.scala:107)
        ... 90 more
```

Though `ST_GeometryType` works fine.
```sql
SELECT ST_GeometryType(ST_GeomFromText('GEOMETRYCOLLECTION EMPTY'));
-- ST_GeometryCollection
```

Edit: I've tried all other empty geometries, and it fails on all of them.

--- a/common/src/test/java/org/apache/sedona/common/FunctionsTest.java
+++ b/common/src/test/java/org/apache/sedona/common/FunctionsTest.java
@@ -3953,6 +3953,28 @@ public class FunctionsTest extends TestBase {
   }
 
   @Test
+  public void geometryTypeWithMeasuredEmpty() {
+    // Regression test for GH-2390: GeometryType fails on all EMPTY geometries
+    assertEquals("POINT", Functions.geometryTypeWithMeasured(GEOMETRY_FACTORY.createEmpty(0)));
+    assertEquals(
+        "LINESTRING", Functions.geometryTypeWithMeasured(GEOMETRY_FACTORY.createLineString()));
+    assertEquals(
+        "POLYGON",
+        Functions.geometryTypeWithMeasured(
+            GEOMETRY_FACTORY.createPolygon(GEOMETRY_FACTORY.createLinearRing())));
+    assertEquals(
+        "MULTIPOINT", Functions.geometryTypeWithMeasured(GEOMETRY_FACTORY.createMultiPoint()));
+    assertEquals(
+        "MULTILINESTRING",
+        Functions.geometryTypeWithMeasured(GEOMETRY_FACTORY.createMultiLineString()));
+    assertEquals(
+        "MULTIPOLYGON", Functions.geometryTypeWithMeasured(GEOMETRY_FACTORY.createMultiPolygon()));
+    assertEquals(
+        "GEOMETRYCOLLECTION",
+        Functions.geometryTypeWithMeasured(GEOMETRY_FACTORY.createGeometryCollection()));
+  }
+
+  @Test
   public void closestPoint() {
     Point point1 = GEOMETRY_FACTORY.createPoint(new Coordinate(1, 1));
     LineString lineString1 =

